pkg/patterns/declarative/reconciler.go

```diff
@@ -96,11 +96,7 @@ func (r *Reconciler) Init(mgr manager.Manager, prototype DeclarativeObject, opts
 	}
 	r.dynamicClient = d
 
-	restMapper, err := apiutil.NewDiscoveryRESTMapper(r.config)
-	if err != nil {
-		return err
-	}
-	r.restMapper = restMapper
+	r.restMapper = mgr.GetRESTMapper()
```

Comment:
Thought: Can we get rid of r.config and the other place we call NewDiscoveryRESTMapper ?